### DISCUSSION TITLE
Proposal: motion for relocating f+ program to L2

### DISCUSSION THREAD

**Discussion by Fatman13** [filecoin-project/notary-governance#804] [General]
Originally posted @ https://github.com/filecoin-project/notary-governance/issues/803

## Simple Summary

Add "explore potentials of moving f+ to L2" to f+ program's OKR.

## Abstract

With FEVM just around the corner, f+ program could finally play around the idea of moving the program to L2. By doing so, it should provide more tools for the team to govern the program in a more transparent and decentralized ways and at the same time lower the complexity of L1 leaving it in a much more cleaner state. Some of the key milestone of this expedition could be the following...

- A whitepaper of f+ program on L2 including mechanics on governance, tokennomics, how program functions on L2 and etc
- Prototyping on FEVM and possibly hold incentivized tests on Calibration
- FIP to remove Datacap entirely on L1

## Motivation

As pointed out by [ZX](https://github.com/filecoin-project/FIPs/discussions/453#discussioncomment-3541311) in Sep last year, the network has succumbed to the macro economic winter. [Network growth](https://github.com/filecoin-project/community/discussions/572) has fallen dramatically with no significant number of new SP joining and existing SP leaving the network. 

By looking at Web2 companies like Meta, it is apparent that **data** is the key to a successful venture. One of the ways to address the aforementioned issue is to onboard more valuable data which in turn increase the value of the whole network. However, f+ program sitting at the core of on-boarding valuable data (as opposed to verified data) as it is structured right now on L1 falls short in accomplishing so from the following aspects...(many of the [issues](https://github.com/filecoin-project/FIPs/issues/203) in Nov, 2021 are still not really addressed)

### 1

From original[ FIP0003](https://github.com/filecoin-project/FIPs/blob/master/FIPS/fip-0003.md)...(Oct, 2020)

> This proposal rebrands the Verified Clients program as Filecoin Plus - to represent the addition of a layer of social trust and all the use cases that will be enabled on the existing Filecoin protocol.

A social layer doesn't scale well and we see how it is played out for **2 years** now. The program tries to adapt over the span of **2 years** but still fails at the goal of distributing **5PiB** valuable data to the network per day. It is clear that the current paradigm does not work in a global scale and need drastic changes like what is proposed here.

### 2 

Current f+ program implicitly encourages division of the community given the current composition of f+ participants. Instead of uniting the community to work toward a common goal, program as it is right now creates a strong "us VS. them" atmosphere crippling data on-boarding efforts. A new governance model based on L2 smart contracts with proper incentive/punitive structure is needed to guide the community to work together.   

### 3

**Valuable data** as opposed to *verified data*. There is very little inherent value of verified data as it does very little to increase the actual value of the network. What the network need is more valuable data determined by free market. To iterate on such model of value evaluation, having f+ on L1 and needing a FIP to adapt is near to impossible. Moving to L2 so that proper incentive could be given to more valuable data will boost the value of the network in orders of magnitudes. 

## Community Highlights

> [sunset of Fil+](https://github.com/filecoin-project/FIPs/discussions/554#discussioncomment-4467725) in favor of L2 mechanisms

> [A better solution](https://github.com/filecoin-project/FIPs/discussions/554#discussioncomment-4376052) is to move Fil+ totally to Layer 2 after FVM is ready.

> [Move f+ off-chain](https://github.com/filecoin-project/community/discussions/572#discussioncomment-4288563) 
  
## Conclusion

Rather than keep building on a skewed foundation, we could argue that the whole premise on which f+ program (FIP0003) is built upon is outdated as the goal of the network **now** is to distribute **5PiB of DC per day**.

- social layer of trust -> should be trustless
- community enforcements -> should be something DAOish
- verified data -> should be valuable data

Moving to L2 may not solve all the problems, but rather the potentials need to be explored. Research efforts need to be directed towards this area so that we all could make an educated decision on if the program should pivot or not. (Or be [terminated](https://filecoinproject.slack.com/archives/C01DLAPKDGX/p1673964538992549?thread_ts=1673963711.017209&cid=C01DLAPKDGX) [completely](https://github.com/filecoin-project/notary-governance/issues/813#issuecomment-1387273258) for good)

**Comment by panges2** [comment]
@Fatman13 I agree with many of the problems you brought up about the current style of Fil+. I do agree that we need many changes going forward. However, I disagree that moving to an L2 will solve this problem. On other chains, the main reason why people create an L2 for a specific use case (e.g. the old dydx on Starkware as perp trading creates high traffic) is to reduce gas costs for the end user as well as reduce traffic on the L1.  

Yes, Fil+ would benefit from moving to an L2 as it is a particular use case that involves many transactions, but the problem will still remain on how to create proper governance. Just because we move to an L2 doesn't mean that we will automatically have better incentive/punitive structures. Furthermore, if we prioritize this movement now, it will add engineering costs. I am not really aware of any fully functional L2 that exists today (both optimistic and zk) for the Filecoin L1. My suggestion is to focus on getting a better incentive/punitive structure while we're on L1 before focusing on moving to L2.

**Reply by Fatman13** [reply]
> I agree with many of the problems you brought up about the current style of Fil+.

Rather than problems, I am arguing that the whole premise on which f+ (FIP0003) is built upon is outdated as the goal of the network **now** is to distribute 5PiB of DC per day.

- social layer of [trust](https://filecoinproject.slack.com/archives/C01DLAPKDGX/p1673964538992549?thread_ts=1673963711.017209&cid=C01DLAPKDGX) -> should be trustless
- community enforcements -> should be something DAOish
- verified data -> should be valuable data

> However, I disagree that moving to an L2 will solve this problem.

I am not saying moving to L2 will solve all problems, but rather the potentials need to be explored. Research efforts need to be directed towards this area so that we could make an educated decision on if the program should pivot or not.

**Reply by herrehesse** [reply]
Very strong points made here @Fatman13 

- social layer of -> should be trustless
- community enforcements -> should be something DAOish
- verified data -> should be valuable data

**Reply by Fatman13** [reply]
> Very strong points made here @Fatman13

Yes, it may sound hash and may very well be wrong, but I still feel strongly that the principles of FIP0003 when it was first designed is not compatible with current network goal of distributing 5PiB DC per day (which is one of f+'s OKR).

**Comment by dkkapur** [comment]
@Fatman13 thanks for sharing this, generally do agree that FVM will unlock some pretty interesting / unique implementations that we can leverage in the Fil+ program. Even without fundamental changes to the program, we could unlock some really cool things like automated notaries / DataCap pipelines for specific types of client applications, etc. 

However - for the points raised, wanted to discuss the following:

> (1) A social layer doesn't scale well and we see how it is played out for 2 years now. The program tries to adapt over the span of 2 years but still fails at the goal of distributing 5PiB valuable data to the network per day. It is clear that the current paradigm does not work in a global scale and need drastic changes like what is proposed here.

Measuring ability to produce and distribute DataCap on a per day basis is not a reasonable metric of success, since Fil+ itself is not driving the funnel itself. The program should instead be measured by (1) how quickly actually trustworthy clients do receive DataCap, and (2) how quickly we can identify cases where the client should not be getting DataCap, and not be supporting them. That means if X PiB of daily demand sourced from trustworthy clients shows up, X PiB should be served with DataCap. A random number that is not actually correlated / aligned with true demand is not meaningful, since DataCap creation and allotment should follow the demand.

On that note, in the last year alone, you've seen this program go from allotting <20 PiB of DataCap to nearly 700 PiBs be given to clients, the number of notaries increase from ~20 to now close to 70+, client addresses go from <1000 to 2500, and serving <2% of active deals to >99%. I think this means we are scaling.

> (2) Current f+ program implicitly encourages division of the community given the current composition of f+ participants. Instead of uniting the community to work toward a common goal, program as it is right now creates a strong "us VS. them" atmosphere crippling data on-boarding efforts. A new governance model based on L2 smart contracts with proper incentive/punitive structure is needed to guide the community to work together.

I strongly believe the us vs. them is a direct outcome of having a network block rewards based incentive, where every verified deal dilutes the ROI of every next verified deal. Unless you have a proposal to fundamentally change the "on-chain subsidy" offered through DataCap and QAP, I don't see how moving to an L2 gets us anything substantial here in the long term. 

Also curious where you see the crippling of onboarding efforts as a result of this. 

> (3) Valuable data as opposed to verified data. There is very little inherent value of verified data as it does very little to increase the actual value of the network. What the network need is more valuable data determined by free market. To iterate on such model of value evaluation, having f+ on L1 and needing a FIP to adapt is near to impossible. Moving to L2 so that proper incentive could be given to more valuable data will boost the value of the network in orders of magnitudes.

Agree with the premise here. We should seek to support and identify truly useful / high leverage use cases of the network. There are several ways to do this, would be interested in seeing a proposal to use a smart contract for this. Other ideas that have been shared are along the lines of additional QAP adjustments for specific subsets of eligible data, i.e., 2*10x for verifiably open/retrievable datasets, etc.

**Reply by Fatman13** [reply]
Just came back from holiday and appreciate the thoughtful reply! @dkkapur 

> Measuring ability to produce and distribute DataCap on a per day basis is not a reasonable metric of successsince, Fil+ itself is not driving the funnel itself.

Maybe I was wrong but in multiple occasions 5Pib DC per day is what I heard to be one of the network OKRs. And it is the easiest indicator for both new and existing network participants to understand the current status of the network. It is also the most fundamental metric of the network since without healthy growth all other metrics are merely smoke in the mirror. As most of the growth is driven solely by DC right now, fil+ program IMHO do share a lot of the responsibilities. Thus using 5Pib DC per day as the target for scaling in my original post. With current staffing of 6 f+ maintainer 90+ notaries, we reached ~2Pib DC per day, imagine 5 or 10Pib DC per day, do we just keep doubling the size of the program?

> I don't see how moving to an L2 gets us anything substantial here in the long term

I am not saying moving to L2 will solve everything but the proposal itself is more about `Add "explore potentials of moving f+ to L2" to f+ program's OKR`. Besides, some could argue that QAP is [distorting the security of L1 consensus](https://github.com/filecoin-project/FIPs/discussions/554#discussioncomment-4772923). 

> Also curious where you see the crippling of onboarding efforts as a result of this.

Filecoin gets compared with AWS a lot. I would say that data onboarding process for filecoin is "like" AWS but with an extra step of f+ program. Most of the community efforts to f+ itself is free but it adds OPex to businesses (both SC and SP). These efforts could be directed to actual data onboarding if we had a more decentralized and trustless f+.

> a proposal to use a smart contract for this

That might be a task for a well funded research team and out of the scope of this `Add "explore potentials of moving f+ to L2" to f+ program's OKR` proposal.

**Comment by PHABRIC77** [comment]
One way of making f+ more programmatic is to insist on paid deals above a certain pay (fiat) threshold. This is the only numerical judge of data quality that I can easily think of …. Challenges around aml/kyc checks ofc-  could a third party be relied on to carry these out? For eg an integrated 3rd party payment platform. Using smart contract is interesting especially if it could help automate sector renewals. Having an automatic mechanism combined with paid deals opens the door to reduce sector durations imo,  something which could help reduce the mismatch between fil availability and pledge deployment. Further, the incentive mechanism for eg multiplier is inherently programmatic and could be set through governance processes not just via market mechanism. Interesting ……

**Reply by Fatman13** [reply]
> One way of making f+ more programmatic is to insist on paid deals above a certain pay (fiat) threshold.

I think people would always find ways to circumvent the KYCs, besides such solution is hard to implement on a global scale with different jurisdiction and lack of trust in the KYC provider in general. If a mandated pay is enforced, I think most likely outcome is that SPs just pay the threshold to their own nodes and get DC.
